net/bluetooth/Kconfig

```diff
@@ -22,9 +22,45 @@ config NET_BLUETOOTH
 
 if NET_BLUETOOTH
 
-config NET_BLUETOOTH_NCONNS
-	int "Max Bluetooth sockets"
+config NET_BLUETOOTH_PREALLOC_CONNS
+	int "Preallocated Bluetooth sockets"
 	default 4
+	---help---
+		Number of Bluetooth connections (all tasks).
+		
+		This number of connections will be pre-allocated during system boot.
+		If dynamic connections allocation is enabled, more connections may
+		be allocated at a later time, as the system needs them. Else this
+		will be the maximum number of connections available to the system
+		at all times.
+
+		Set to 0 to disable (and rely only on dynamic allocations).
+
+config NET_BLUETOOTH_ALLOC_CONNS
+	int "Dynamic Bluetooth connections allocation"
+	default 0
+	---help---
+		Dynamic memory allocations for Bluetooth.
+
+		When set to 0 all dynamic allocations are disabled.
+
+		When set to 1 a new connection will be allocated every time,
+		and it will be free'd when no longer needed.
+
+		Setting this to 2 or more will allocate the connections in
+		batches (with batch size equal to this config). When a
+		connection is no longer needed, it will be returned to the
+		free connections pool, and it will never be deallocated!
+
+config NET_BLUETOOTH_MAX_CONNS
+	int "Maximum number of Bluetooth connections"
+	default 0
```

Comment:
ditto

Reply:
Done.

Reply:
but no change

Reply:
Isn't this OK?  
https://github.com/apache/nuttx/pull/7525/commits/d550e68264632321f12ccae5b28a24ae6da6586e#diff-789fc80906d45426aebac87a9b963c00ea7679ebe7155dbe2d8e7015e81f6ac3R58

Reply:
Yes.